### PR TITLE
ci: Update restore-node dirty check to ignore untracked endo-sha.txt

### LINKED ISSUE #9708: Custom GitHub action "restore-node" can fail on job retry
## Describe the bug
Our custom [restore-node GitHub action](https://github.com/Agoric/agoric-sdk/blob/bd50e5e0a7ed5ce1da8faf901c28ea4328bbe84c/.github/actions/restore-node/action.yml) captures the relevant endo branch in local file **endo-sha.txt** (which is also staged in git), then associates the agoric-sdk directory with an [`@actions/cache`](https://github.com/actions/cache) entry keyed in part upon that branch, then executes `yarn install` and `yarn build` only on cache miss, then finally does a "dirty check" that tolerates staged files (cf. #8168) but not untracked files. However, if the cache was populated by a job that removed its own **endo-sha.txt** (_such as workflow integration "getting-started" with matrix.cli value "registry/yarn", "registry/npm", or "registry/npx", all of which include a ["Start local NPM registry" step](https://github.com/Agoric/agoric-sdk/blob/bd50e5e0a7ed5ce1da8faf901c28ea4328bbe84c/.github/workflows/integration.yml#L84-L90) that executes `git reset --hard HEAD`_), then the restored git directory will be unaware of the local file and report it as untracked, failing the action and its containing job.

This happened in the CI for #9688, where:
1. **getting-started (registry/yarn)** won the race to save the cache entry
   * _Start local NPM registry_
     1. [<tt>git reset --hard HEAD</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27392837141#step:4:252)
   * _Post Run /./.github/actions/restore-node_
     1. [<tt>Cache saved with key: Linux-X64-22.04-node-18.18-built-0-c7d1c9f2923282881ba8d912fc314659e5e5a581-NOPE</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27391851906#step:15:6)
3. **getting-started (link-cli/yarn)** failed due to what looks like flakiness (#9710)
   * _run agoric-cli integration-test_
     1. [[agd tx gov deposit] <tt>Error: rpc error: code = Unknown desc = rpc error: code = Unknown desc = **account sequence mismatch, expected 63, got 62**: incorrect account sequence [agoric-labs/cosmos-sdk@v0.46.16-alpha.agoric.2.4/x/auth/ante/sigverify.go:269] With gas wanted: '18446744073709551615' and gas used: '38588' : unknown request</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27391851751#step:7:951)
4. a retry of **getting-started (link-cli/yarn)** failed in restore-node
   * _Run /./.github/actions/restore-node_
     1. [<tt>Cache restored from key: Linux-X64-22.04-node-18.18-built-0-c7d1c9f2923282881ba8d912fc314659e5e5a581-NOPE</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27392837141#step:4:226)
     2. …
     3. [<tt>'[' -n '?? endo-sha.txt' ']'</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27392837141#step:4:243) (_from `if [ -n "$(git status --porcelain | grep -Eve '^[^ ]  '; true)" ]; then … exit 1`_)
     5. …
     6. [<tt>Untracked files: … endo-sha.txt</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27392837141#step:4:246)
     7. …
     8. [<tt>Unexpected dirty git status</tt>](https://github.com/Agoric/agoric-sdk/actions/runs/9913890868/job/27392837141#step:4:252)

## Expected behavior
restore-node should not expect that the endo-sha.txt file it creates is guaranteed to be included in a cache entry created at the end of its containing job. Specifically, I think it should tolerate absence of the file in its "dirty check" step.

### PR DESCRIPTION
**Pull request by gibson042**
Fixes #9708

## Description
Tolerates removal of endo-sha.txt by e.g. `git reset --hard HEAD` contributing to a restored cache state. Also includes minor cleanup of GitHub actions files.

### Security Considerations
No relevant change.

### Scaling Considerations
n/a

### Documentation Considerations
n/a

### Testing Considerations
It's difficult to test these kinds of changes, but the proof of the pudding is in the taste.

### Upgrade Considerations
n/a

### CODE REVIEW

File: .github/actions/restore-node/action.yml [hunk 1]
```diff
@@ -156,28 +168,31 @@ runs:
...
+        # Fail if `git status` reports anything other than the following:
+        # * an untracked endo-sha.txt from above
+        # * work tree files that have been staged without further changes
+        #   (e.g., package.json or yarn.lock) as indicated by the Y position
+        #   in "XY PATH" being a space
+        if git status --porcelain | awk '$0 !~ /^[?][?] endo-sha.txt$|^. |^$/ { exit 1; }'; then
           git status
           echo "Unexpected dirty git status" 1>&2
           exit 1
         fi
```
Comment (michaelfig) [inline .github/actions/restore-node/action.yml]: The logic of the `if` condition is backwards here:
```suggestion
        if git status --porcelain | awk '$0 !~ /^[?][?] endo-sha.txt$|^. |^$/ { exit 1; }'; then :
        else
          git status
          echo "Unexpected dirty git status" 1>&2
          exit 1
        fi
```

**Review by michaelfig — COMMENTED** [review]
Looking good.  I'd like to see CI pass before I approve.

File: .github/actions/restore-node/action.yml (hunk 1 above)
Comment (gibson042) [reply .github/actions/restore-node/action.yml]: D'oh, nice catch! I switched back to the old approach with the new grep pattern.

**Review by michaelfig — APPROVED** [review]

**Comment by LuqiPan** [comment]
@gibson042 any plan to land this PR today/this week? Looks like there are a few victims to this CI bug.

**Pull request merged**